url8af84d296f_mrcosta_gantt_generator_tgz-pJ8-org_gantt_generator_mrcpsp_RunJ8/src/org/swiftgantt/ui/task/DefaultTaskRenderer.java

```diff
@@ -0,0 +1,379 @@
+		int endX = pred.calcTaskEndPointX(startX);
+		if (TaskHelper.isAllowAccurateTaskBar(tu)) {
+			endX -= (1 - pred.getEndTimeExcees(tu)) * super.step_length;// For accurate task bar.
+		}
+		int y = pred.calcTaskPointY(rect, predIndex, padding_v);
+		if (logger.isDebugEnabled()) {
+			logger.debug("#########" + rect + " " + predIndex + " " + padding_v);
+		}
+		Point ps = new Point(endX, y + super.task_bar_height / 2);//Start point from predecessor.
+		Point p1 = new Point(endX + offset, ps.y);//1st corner point
+		Point p2 = new Point(p1.x, ps.y + super.row_height / 2);//2ed corner point
+		Point p3 = new Point(endX - offset, p2.y);// 3rd corner point
+		Point p4 = new Point(p3.x, y + super.task_bar_height / 2 + (thisIndex - predIndex) * super.row_height);//4th corner point
+		Point pe = new Point(endX, p4.y);// End point at current task bar.
+		g.drawLine(ps.x, ps.y, p1.x, p1.y);
+		g.drawLine(p1.x, p1.y, p2.x, p2.y);
+		g.drawLine(p2.x, p2.y, p3.x, p3.y);
+		g.drawLine(p3.x, p3.y, p4.x, p4.y);
+		g.drawLine(p4.x, p4.y, pe.x, pe.y);
+		g.drawLine(pe.x - 5, pe.y + 3, pe.x, pe.y);
+		g.drawLine(pe.x - 5, pe.y - 3, pe.x, pe.y);
+	}
+
+	/*
+	 * @param distance Distance from predecessor's end to this tasks start.
+	 */
+	private void drawDistantConnectToPredecessor(TaskWrapper t, TaskWrapper pred, float distance) {
+		TimeUnit tu = gantt.getTimeUnit();
+		int offset = 10; // Offset for connect lines.
+		int thisIndex = tasks.indexOf(t);
+		int predIndex = tasks.indexOf(pred);
+		int startPX = pred.calcTaskStartPointX(rect, gantt.getTimeUnit(), gantt.getModel().getKickoffTime());
+		int endX = pred.calcTaskEndPointX(startPX);
+		if (TaskHelper.isAllowAccurateTaskBar(tu)) {
+			endX -= (1 - pred.getEndTimeExcees(tu)) * super.step_length;// For accurate task bar.
+		}
+		int y = pred.calcTaskPointY(rect, predIndex, padding_v);
+		Point ps = new Point(endX, y + super.task_bar_height / 2);
+		Point pe = new Point(endX + (int) (distance * super.step_length), y + super.task_bar_height / 2 + (thisIndex - predIndex) * super.row_height);
+		Point p1 = null, p2 = null;
+		if (this.connectorStyle == 1) {
+			p1 = new Point(ps.x + offset, ps.y);
+			p2 = new Point(p1.x, pe.y);
+		} else if (this.connectorStyle == 2) {
+			p1 = new Point(pe.x - offset, ps.y);
+			p2 = new Point(p1.x, pe.y);
+		}
+		g.drawLine(ps.x, ps.y, p1.x, p1.y);
+		g.drawLine(p1.x, p1.y, p2.x, p2.y);
+		g.drawLine(p2.x, p2.y, pe.x, pe.y);
+
+		//Draw Arrow.
+		g.drawLine(pe.x - 5, pe.y + 3, pe.x, pe.y);
+		g.drawLine(pe.x - 5, pe.y - 3, pe.x, pe.y);
+	}
+
+	/**
+	 * Leaf task is the lowest level task in the task tree.
+	 * @param x
+	 * @param y
+	 * @param steps
+	 * @param preVacancy Pre vacancy before first full step.
+	 * @param seqExcess Sequent excess after last full step.
+	 * @param task
+	 */
+	private void drawLeafTask(int x, int y, int taskSteps, float preVacancy, float seqExcess, TaskWrapper tw) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 318-340, lines 343-349 into separate methods.